src/App.jsx

```diff
@@ -14,33 +15,104 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App = () => (
-  <div className="section content">
-    <div className="buttons">
-      <button type="button" className="button is-info is-light">
-        Sort alphabetically
-      </button>
+const getSortedGoods = (goods, sortType, reversed) => {
+  const sortedGoods = [...goods];
 
-      <button type="button" className="button is-success is-light">
-        Sort by length
-      </button>
+  sortedGoods.sort((a, b) => {
+    switch (sortType) {
+      case 'alphabetically':
+        return a.localeCompare(b);
+      case 'length':
+        return a.length - b.length;
+      default:
+        return 0;
+    }
+  });
 
-      <button type="button" className="button is-warning is-light">
-        Reverse
-      </button>
+  if (reversed) {
+    sortedGoods.reverse();
+  }
 
-      <button type="button" className="button is-danger is-light">
-        Reset
-      </button>
-    </div>
+  return sortedGoods;
+};
+
+export const App = () => {
+  const [goods, setGoods] = useState(goodsFromServer);
+  const [originalGoods] = useState([...goodsFromServer]);
+  const [reversed, setReversed] = useState(false);
+  const [activeButton, setActiveButton] = useState('');
+
+  const sortGoods = type => {
+    const sortedGoods = getSortedGoods(originalGoods, type, reversed);
+
+    setGoods(sortedGoods);
+    setActiveButton(type);
+  };
+
+  const toggleReverse = () => {
+    const sortedGoods = getSortedGoods(goods, activeButton, !reversed);
+
+    setGoods(sortedGoods);
+    setReversed(!reversed);
+  };
+
+  const resetGoods = () => {
+    setGoods(originalGoods);
+    setReversed(false);
+    setActiveButton('');
+  };
```

Comment:
```suggestion
  const sortGoods = type => {
    setActiveButton(type);
  };

  const toggleReverse = () => {
    setReversed(!reversed);
  };

  const resetGoods = () => {
    setReversed(false);
    setActiveButton('');
  };
  
  const sortedGoods = getSortedGoods(originalGoods, activeButton, reversed);
```
activeButton is better to rename to sortValue or smth like that